apps/google-analytics-4/src/components/FormControlServiceAccountKey.tsx

```diff
@@ -186,33 +188,37 @@ const FormControlServiceAccountKey = ({
   };
 
   const formControl = (
-    <FormControl marginTop="spacingL" id="accountCredentialsFile">
-      <FormControl.Label>Google Service Account Key File</FormControl.Label>
+    <FormControl
+      marginTop="spacingL"
+      id="accountCredentialsFile"
+      isInvalid={formControlState.isInvalid}
+      isRequired={isRequired}
+    >
+      <FormControl.Label isRequired={isRequired}>Google Service Account Key File</FormControl.Label>
       <Textarea
         name="accountCredentialsFile"
         placeholder={placeholderText}
         rows={10}
         className={styles.credentialsInput}
         value={serviceAccountKeyFile}
         onChange={onKeyFileTextareaChange}
-        isInvalid={formControlState.isInvalid}
       />
       {formControlState.isInvalid ? (
         <FormControl.ValidationMessage>
           Error: {formControlState.errorMessage}
         </FormControl.ValidationMessage>
+      ) : serviceAccountKeyFile ? (
+        <Flex marginTop="spacingXs" alignItems="center">
+          <Flex isInline={true} alignItems="center">
+            <CheckCircleIcon variant="positive" />{' '}
+            <Text as="p" marginLeft="spacing2Xs" fontColor="gray700">
```

Comment:
If you make this a `p` why not use the `Paragraph ` element from forma directly? 🤔

Reply:
Actually this is similar to what I explained with regards to @andipaetzold 's comment.

The idea here is that we flip between the "valid" state with a green check icon and the message and the "invalid" state with error icon and similar message.

The error message we get for free with Forma36.

The "valid" message isn't provided, so I just took the lower level components from here https://github.com/contentful/forma-36/blob/main/packages/components/forms/src/ValidationMessage/ValidationMessage.tsx#L40-L45 more or less verbatim. This also helped have a smooth transition between the two states, whereas `Paragraph` introduces extra margin which would need to be fixed.

In a perfect world our component library would provide something like this perhaps. 😄